### PR TITLE
Use swift-format

### PR DESCRIPTION
**Pull request by MahdiBM**

### CODE REVIEW

**Comment by MahdiBM** [comment]
This is just the first run, I don't necessarily agree with all of this.

Specifically I don't like all the guards (there an option to disable that luckily) and u also don't like how it adds an extra line after return.

File: .swift-format [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "UseLetInEveryBoundCaseVariable": false,
+    "UseShorthandTypeNames": true,
+    "UseSingleLinePropertyGetter": false,
+    "UseSynthesizedInitializer": true,
+    "UseTripleSlashForDocumentationComments": true,
+    "UseWhereClausesInForLoops": false,
+    "ValidateDocumentationComments": false
+  },
+  "spacesAroundRangeFormationOperators": false,
+  "tabWidth": 8,
```
Comment (gwynne) [inline .swift-format:56]:
```suggestion
  "tabWidth": 4,
```

File: .swift-format [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "FullyIndirectEnum": true,
+    "GroupNumericLiterals": true,
+    "IdentifiersMustBeASCII": true,
+    "NeverForceUnwrap": false,
+    "NeverUseForceTry": false,
+    "NeverUseImplicitlyUnwrappedOptionals": false,
+    "NoAccessLevelOnExtensionDeclaration": false,
+    "NoAssignmentInExpressions": true,
+    "NoBlockComments": true,
+    "NoCasesWithOnlyFallthrough": true,
```
Comment (gwynne) [inline .swift-format:35]:
```suggestion
    "NoCasesWithOnlyFallthrough": false,
```

File: .swift-format [hunk 3]
```diff
@@ -0,0 +1,58 @@
+{
+  "fileScopedDeclarationPrivacy": {
+    "accessLevel": "private"
+  },
+  "indentation": {
+    "spaces": 4
+  },
+  "indentConditionalCompilationBlocks": false,
+  "indentSwitchCaseLabels": false,
+  "lineBreakAroundMultilineExpressionChainComponents": true,
```
Comment (gwynne) [inline .swift-format:10]:
```suggestion
  "lineBreakAroundMultilineExpressionChainComponents": false,
```

File: .swift-format [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  "lineBreakBeforeControlFlowKeywords": false,
+  "lineBreakBeforeEachArgument": true,
+  "lineBreakBeforeEachGenericRequirement": true,
+  "lineLength": 100,
```
Comment (gwynne) [inline .swift-format:14]:
```suggestion
  "lineLength": 120,
```

File: .swift-format [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    "AllPublicDeclarationsHaveDocumentation": false,
+    "AlwaysUseLowerCamelCase": false,
+    "AmbiguousTrailingClosureOverload": true,
+    "BeginDocumentationCommentWithOneLineSummary": false,
+    "DoNotUseSemicolons": true,
+    "DontRepeatTypeInStaticProperties": false,
+    "FileScopedDeclarationPrivacy": true,
+    "FullyIndirectEnum": true,
+    "GroupNumericLiterals": true,
+    "IdentifiersMustBeASCII": true,
```
Comment (gwynne) [inline .swift-format:28]:
```suggestion
    "IdentifiersMustBeASCII": false,
```

**Comment by MahdiBM** [comment]
I should also set up some CI. Not sure if in this PR.

**Comment by gwynne** [comment]
> I should also set up some CI. Not sure if in this PR.

CI to do what?

**Comment by gwynne** [comment]
If it works out well for Penny, I'll add formatting to the shared CI workflows

File: Lambdas/AutoFaqs/S3AutoFaqsRepository.swift [hunk 6]
```diff
@@ -53,10 +56,13 @@ public struct S3AutoFaqsRepository {
...
-                "response-body": .string(String(buffer: body)),
-                "error": "\(error)"
-            ])
+            logger.error(
+                "Cannot find any data in the bucket",
+                metadata: [
+                    "response-body": .string(String(buffer: body)),
+                    "error": "\(error)",
+                ]
+            )
```
Comment (gwynne) [inline Lambdas/AutoFaqs/S3AutoFaqsRepository.swift:65]: I am really, really, _really_ not a fan of all these extra newlines. Would setting `lineBreakBeforeEachGenericRequirement` and `lineBreakBeforeEachArgument` to `false` make it ease up on this?

File: Lambdas/GHHooks/EventHandler/Handlers/DocsIssuer.swift [hunk 7]
```diff
@@ -64,32 +65,38 @@ struct DocsIssuer {
...
+                    owner: repo.owner.login,
+                    repo: repo.name,
+                    commit_sha: commitSHA
+                )
+            )
         )
 
         guard case let .ok(ok) = response,
-              case let .json(json) = ok.body else {
+            case let .json(json) = ok.body
```
Comment (gwynne) [inline Lambdas/GHHooks/EventHandler/Handlers/DocsIssuer.swift:78]: While I approve of it moving the `else` to its own line, I actively dislike it actively un-lining-up the individual conditions.

File: Lambdas/GHHooks/Extensions/Embed+Equtable.swift [hunk 8]
```diff
@@ -2,64 +2,47 @@ import DiscordModels
...
-        lhs.image == rhs.image &&
-        lhs.thumbnail == rhs.thumbnail &&
-        lhs.video == rhs.video &&
-        lhs.provider == rhs.provider &&
-        lhs.author == rhs.author &&
-        lhs.fields == rhs.fields
+        lhs.title == rhs.title && lhs.type == rhs.type && lhs.description == rhs.description && lhs.url == rhs.url
+            && lhs.timestamp?.date == rhs.timestamp?.date && lhs.color == rhs.color && lhs.footer == rhs.footer
+            && lhs.image == rhs.image && lhs.thumbnail == rhs.thumbnail && lhs.video == rhs.video
+            && lhs.provider == rhs.provider && lhs.author == rhs.author && lhs.fields == rhs.fields
```
Comment (gwynne) [inline Lambdas/GHHooks/Extensions/Embed+Equtable.swift:8]: This also is way too aggressive on the formatter's part, IMO

**Review by gwynne — CHANGES_REQUESTED** [review]
I have to say, this kind of thing is the one area where `swiftformat` is still much superior to `swift-format` - Nick Lockwood's Swift parser may be abysmally bug-ridden and incapable of actually understanding what it's doing (really it's just lexing, not parsing - and poorly at that), but it has _considerably_ more numerous and precise knobs for controlling the behavior.

File: Lambdas/AutoFaqs/S3AutoFaqsRepository.swift (hunk 6 above)
Comment (MahdiBM) [reply Lambdas/AutoFaqs/S3AutoFaqsRepository.swift:65]: I did try some combinations of disabling the "lineBreak" settings, but didn't try just disabling all of them at once.

**Comment by MahdiBM** [comment]
I don't think swift-format is useable in its current state. I might or might not try the same with Nick Lockwood's SwiftFormat. That at least works nicely most of the time, although has some bugs.

**Pull request closed without merging**